BrainPortal/app/controllers/userfiles_controller.rb

```diff
@@ -188,6 +188,29 @@ def create_parent_child #:nodoc:
     redirect_to :action => :index
   end
 
+  # Use the rest of the route as a path into a file collection.
+  # [:file_path] a path relative to the file collection's route directory.
+  #
+  # GET /userfiles/1/file_collection_content/path/to/file/in/file_collection
+  def file_collection_content
+    @userfile = Userfile.find_accessible_by_user(params[:id], current_user, :access_requested => :read)
+    if @userfile.nil?
+      raise ActiveRecord::RecordNotFound
+    end
+    path = @userfile.cache_full_path.to_s + '/' + params[:file_path]
+    if params[:format]
+      path += '.' + params[:format]
+    end
+    if not File.file?(path)
```

Comment:
I think this will fail (maybe) if the file is a symbolic link within the collection (check the doc for `File.file?()`

Reply:
Also, make a check that the file path doesn't go outside the collection (there are methods in Pathname to help make sure the final destination of a path or symlink ends up within a certain directory tree)

Reply:
it should be fine: https://ruby-doc.org/core-2.2.0/File.html#method-c-file-3F

Reply:
ah good point!

Reply:
In case of violation, maybe `head :unauthorized`

Reply:
I have made the changes! I expand the path from read_link and check if it starts with the absolute path of the userfile.